### PR TITLE
Cannot run on Windows

### PR DESCRIPTION
**Pull request by danepowell**
This is just a start. It converts the sshDir, but we need to convert nearly _all_ of the parameters (anything that depends on `$_ENV['HOME']`). I'm not sure how to feel about this approach. Reducing the number of service parameters / dependencies injected is good I think. Feels weird to kill all parameter though.

The reason this is so complicated: we use the home directory in our parameters to set other things like the data and ssh dirs, but HOME isn't set on Windows, it's some other variable. You can't use expressions in parameters, they have to be static, so we can't dynamically set the home directory.

I think we'll need to strip out most of the parameters and convert them into methods on the LocalMachineHelper, and then pass the LocalMachineHelper as a service and call its methods to get the values instead of getting them as parameters.

Alternatively, we might be able to bind these paths instead of setting them as parameters. That would involve a lot less refactoring. But the static methods can't be tested as easily, we'd need new tests.

### CODE REVIEW

**Comment by danepowell** [comment]
I think this is the right approach but only the home directory should be mutable, everything else should be relative to that. Will need to adjust tests to accommodate that.

**Comment by danepowell** [comment]
@imnetworku since you've worked with Symfony service containers before, does this approach make sense to you?

To recap: we store some config in the user's home directory, such as `~/.ssh`, `~/.acquia/cloud_api.conf`. We previously defined these paths via `config/services.yml` based on the value of the `$HOME` environment variable. The problem is that Windows doesn't always have HOME, so we need to set the HOME directory dynamically.

The problem is that you can't set a service parameter dynamically, and almost _all other parameters_ depend on it. So I think they can no longer be parameters at all. I was going to create methods on LocalMachineHelper to set each of them dynamically.

**Comment by imnetworku** [comment]
@danepowell That is one possibility, but I think there could be a workaround to make it seem like the service parameters are set dynamically. Let me do a couple PoCs and get back to you
> @imnetworku since you've worked with Symfony service containers before, does this approach make sense to you?
> 
> To recap: we store some config in the user's home directory, such as `~/.ssh`, `~/.acquia/cloud_api.conf`. We previously defined these paths via `config/services.yml` based on the value of the `$HOME` environment variable. The problem is that Windows doesn't always have HOME, so we need to set the HOME directory dynamically.
> 
> The problem is that you can't set a service parameter dynamically, and almost _all other parameters_ depend on it. So I think they can no longer be parameters at all. I was going to create methods on LocalMachineHelper to set each variable (no longer parameters) dynamically. Then services will need to call `LocalMachineHelper::getSshDir()`, `LocalMachineHelper::getDataDir()`, and such as appropriate rather than receiving a parameter.

**Comment by imnetworku** [comment]
Picking up for manual test

**Comment by imnetworku** [comment]
Manual Test Failed.

In the left side is the failing test on Windows 10 without HOME environment variable defined. After the failure, tested again with HOME system environment variable created and it works.

image

**Comment by imnetworku** [comment]
Manual Test Failed.

Without an explicitly defined HOME environment variable, the output is the same.
image

File: config/services.yml [hunk 1]
```diff
@@ -3,7 +3,6 @@ parameters:
   env(ACLI_REPO_ROOT): "%kernel.project_dir%"
   app.data_dir: "%env(HOME)%/.acquia"
```
Comment (imnetworku) [inline config/services.yml:4]: @danepowell This is the line that is causing the failure.

**Comment by imnetworku** [comment]
@danepowell I went through the transitive changes needed to get it to work on windows... it works but i'm not sure how elegant you think it is... 
https://github.com/imnetworku/cli/commit/c0c065c728d111441005d80e588dea9445e2c389
image

**Comment by danepowell** [comment]
Thanks, I'll check your feature branch. The alternative is that we call LocalMachineHelper statically to find and set the HOME environment variable (if it's not already set) in bin/acli, before the service container gets compiled.

**Comment by danepowell** [comment]
I think there might still be value in what I attempted here (consolidating parameters and arguments into LocalMachineHelper), but it's not necessary to fix the issue at hand (Windows Powershell). See #187 for a much simpler approach.

**Pull request closed without merging**